### ISSUE TITLE
Dynamic "title" for each option

### ISSUE THREAD

**Issue by Kharkare** [leocaseiro/angular-chosen#224]
I have an array of objects which I am populating in my select box.

```
$scope.options = [{
    	"name" : "Option 1",
      "value" : "Value1",
      "tooltiptext" : "Option 1 text"
    },
    {
    	"name" : "Option 2",
      "value" : "Value2",
      "tooltiptext" : "Option 2 text"
    },
    {
    	"name" : "Option 3",
      "value" : "Value3",
      "tooltiptext" : "Option 3 text"
    },
    {
    	"name" : "Option 4",
      "value" : "Value4",
      "tooltiptext" : "Option 4 text"
    }];
```
I want to show the tooltip corresponding to the selected option on mouse hover over the select box. I am using `ng-attr-title` to display the tool tip text

This works if I use simple select box, but doesn't work with angular-chosen.
[Here](https://jsfiddle.net/Kunalh/snzecjyz/9/) is a working example of the above code

How can I make it work with angular-chosen?

**Comment by leocaseiro** [comment]
Hi @Kharkare, thank you for posting your issue.

I believe it's a chosen [issue](https://github.com/harvesthq/chosen/issues/1095). Also the [stackoverflow](http://stackoverflow.com/questions/14569903/how-to-add-title-attribute-to-chosen-option-tags) solution doesn't seems to fix very well.


Another fix that a hack to re-compile the directive to chosen would fix (I have a fake update with $timeout and ngIf).

I'm thinking in working in a update emit for the directive that could be useful for many openned issues. It might fix this one as well.

Atm, you could use this solution, that I don't like, but temporary could work:

```html
<select ng-change="updatechosen()" ng-if="show" ng-model="selected.chosen" chosen ng-options="opt.value as opt.name for opt in options"
    ng-attr-title="{{selectedChosenOptionObject().tooltiptext}}" style="width:155px;">
    <option value=""></option>
</select>
```

```javascript
$scope.show = true;
$scope.updatechosen = function() {
    $scope.show = false;
    $timeout(function() {
        $scope.show = true;
    });
}
```

PS: Also, make sure you always use [an object for ng-model](http://stackoverflow.com/questions/17606936/why-dont-the-angularjs-docs-use-a-dot-in-the-model-directive/17607794#17607794)

See demo working on plunker: http://plnkr.co/edit/Gt6uF9QfulNp6UzpzOrG?p=preview

**Comment by Kharkare** [comment]
Hi @leocaseiro , thank you for your answer.

Shall use the solution suggested by you. This would work for now.
Waiting for the update to release.

**Comment by Ankushdaharwal** [comment]
hii @leocaseiro 

I have tried this solution too but i want title when the options are available  

link you can see i have selected option 3 but the title tag is showing option 1

you can check your code

**Comment by VanTanev** [comment]
@Ankushdaharwal Your comment is not intelligible, can you rephrase it?

Are you interested in doing a dynamic title for the chosen? If so, you should look into writing a custom directive on the `<select>` that hooks into the `change` event, and then set the title by whatever means you want - `ng-attr-title` or directly `jQuery.attr`.

**Comment by Ankushdaharwal** [comment]
hii @VanTanev 

 [https://stackoverflow.com/questions/54827504/how-can-i-add-mouseover-in-ng-option-tag](url)

you can visit this link for my doubt 

and can u please share me code for your suggestion??

**Comment by VanTanev** [comment]
Not an `angular-chosen` issue. Closing.
